src/App.jsx

```diff
@@ -10,68 +10,60 @@ import "./styles/App.css";
 function App() {
   const [contacts, setContacts] = useState([]);
   const [posts, setPosts] = useState([]);
-  const [postComments, setPostComments] = useState([]);
+  const [postComments, setPostComments] = useState({});
   const [newPost, setNewPost] = useState("");
   const [commentInputs, setCommentInputs] = useState({});
   const { postId } = useParams();
   const reversedPostData = [...posts].reverse();
 
   useEffect(() => {
     fetch("https://boolean-api-server.fly.dev/Callumhayden99/contact")
-      .then((response) => response.json())
-      .then((data) => setContacts(data))
-      .catch((error) => console.error("Error fetching contacts:", error));
+      .then(response => response.json())
+      .then(data => setContacts(data))
+      .catch(error => console.error("Error fetching contacts:", error));
 
     fetch("https://boolean-api-server.fly.dev/Callumhayden99/post")
-      .then((response) => response.json())
-      .then((data) => setPosts(data))
-      .catch((error) => console.error("Error fetching posts:", error));
-  }, []);
+      .then(response => response.json())
+      .then(data => setPosts(data))
+      .catch(error => console.error("Error fetching posts:", error));
 
-  useEffect(() => {
-    contacts.forEach((contact) => {
-      if (!postComments[contact.id]) {
-        fetch(
-          `https://boolean-api-server.fly.dev/Callumhayden99/post/${contact.id}/comment`
-        )
-          .then((response) => response.json())
-          .then((data) => {
-            setPostComments((prevData) => ({
-              ...prevData,
-              [contact.id]: data,
-            }));
-          })
-          .catch((error) =>
-            console.error("Error fetching post comments:", error)
-          );
-      }
+    posts.forEach(post => {
+      fetch(`https://boolean-api-server.fly.dev/Callumhayden99/post/${post.id}/comment`)
+        .then(response => response.json())
+        .then(data => {
+          setPostComments(prevData => ({
+            ...prevData,
+            [post.id]: data,
+          }));
+        })
+        .catch(error => console.error(`Error fetching comments for post ${post.id}:`, error));
     });
-  }, [contacts, postComments]);
+  }, [posts]);
```

Comment:
I think this could result in an infinite loop of fetch requests, since the post variable would be updated, trigger the  useEffect, be updated again, and so on...